package.json

```diff
@@ -27,8 +27,7 @@
 		"./assets/js/blocks/filter-wrapper/register-components.ts",
 		"./assets/js/blocks/product-query/variations/**.tsx",
 		"./assets/js/blocks/product-query/index.tsx",
-		"./assets/js/blocks/product-query/inspector-controls.tsx",
-		"./assets/js/templates/revert-button/index.tsx"
```

Comment:
Since `revert-button/index.tsx` is the only file that has a `@wordpress/edit-site` dependency. I suspect this is actually the cause of the issue, especially since it is [imported as a root asset](https://github.com/woocommerce/woocommerce-blocks/blob/2cddfcdc3001afd412450809fbc48a11046b90f9/assets/js/index.js#L8)! It should only be imported/utilized in the site editor environment.

Reply:
> Since revert-button/index.tsx is the only file that has a @wordpress/edit-site dependency. I suspect this is actually the cause of the issue, especially since it is [imported as a root asset](https://github.com/woocommerce/woocommerce-blocks/blob/2cddfcdc3001afd412450809fbc48a11046b90f9/assets/js/index.js#L8)! It should only be imported/utilized in the site editor environment.

Oh, nice catch: that sounds like the cause of the problem here indeed!

> What was this path removed from the side-effects configuration?

The revert button works as expected with and without this side-effect: I'm unsure about its role in our codebase. Tagging @gigitux for confirmation.

Reply:
Thanks for working on this!

> Since revert-button/index.tsx is the only file that has a @wordpress/edit-site dependency. I suspect this is actually the cause of the issue, especially since it is [imported as a root asset](https://github.com/woocommerce/woocommerce-blocks/blob/2cddfcdc3001afd412450809fbc48a11046b90f9/assets/js/index.js#L8)! It should only be imported/utilized in the site editor environment.

I was aware of this issue, I tried to use a dynamic import, but it didn't work. [I wrote a message on the Gutenberg PR.](https://github.com/WordPress/gutenberg/pull/50257#issuecomment-1549279797). I have no idea how to fix this in a good way, but I'm happy to help if needed!

> The revert button works as expected with and without having "./assets/js/templates/revert-button/index.tsx" in this side-effect: it looks like this was added so it can run on production builds. Tagging @gigitux for confirmation.

True!

Reply:
> The revert button works as expected with and without having "./assets/js/templates/revert-button/index.tsx" in this side-effect: it looks like this was added so it can run on production builds. Tagging @gigitux for confirmation.

That's odd, the described behaviour in this PR of the navigation block issue being resolved in production builds but _not_ in development builds is consistent with the import being stripped out of production builds because it is considered a side-effect. This should be easily verifiable by checking to see if `wp.editSite` is available in the post editor with a production build of Woo Blocks.

Reply:
You are right; I may have hit caching on my initial tests. I just had an idea to circumvent this: I will try to conditionally load the necessary script via PHP rather than JS, as I suspect it's not possible to truly conditionally import a module directly in the JavaScript execution flow.